gluoncv/utils/viz/image.py

```diff
@@ -38,3 +38,83 @@ def plot_image(img, ax=None, reverse_rgb=False):
+                  left_txt_list=None, left_txt_corner=(10, 150),
+                  title_txt_list=None, title_txt_corner=(500, 50),
+                  canvas_name='demo'):
+    """Visualize image with OpenCV.
+
+    Parameters
+    ----------
+    img : numpy.ndarray or mxnet.nd.NDArray
+        Image with shape `H, W, 3`.
+    scale : float
+        The scaling factor of the output image
+    upperleft_txt : str, optional, default is None
+        If presents, will print the string at the upperleft corner
+    upperleft_txt_corner : tuple, optional, default is (10, 100)
+        The bottomleft corner of `upperleft_txt`
+    left_txt_list : list of str, optional, default is None
+        If presents, will print each string in the list close to the left
+    left_txt_corner : tuple, optional, default is (10, 150)
+        The bottomleft corner of `left_txt_list`
+    title_txt_list : list of str, optional, default is None
+        If presents, will print each string in the list close to the top
+    title_txt_corner : tuple, optional, default is (500, 50)
+        The bottomleft corner of `title_txt_list`
+    canvas_name : str, optional, default is 'demo'
+        The name of the canvas to plot the image
+
+    Examples
+    --------
+
+    from matplotlib import pyplot as plt
+    ax = plot_image(img)
+    plt.show()
+    """
+    from ..filesystem import try_import_cv2
+    cv2 = try_import_cv2()
+
+    height, width, _ = img.shape
+    img = cv2.resize(img, (int(width * scale), int(height * scale)))
+    if upperleft_txt is not None:
+        font = cv2.FONT_HERSHEY_SIMPLEX
+        bottomLeftCornerOfText = upperleft_txt_corner
+        fontScale = 1
+        fontColor = (255, 255, 255)
+        thickness = 3
+
+        cv2.putText(img, upperleft_txt, bottomLeftCornerOfText,
+                    font, fontScale, fontColor, thickness)
+
+    if left_txt_list is not None:
+        starty = left_txt_corner[1]
+        for txt in left_txt_list:
+            font = cv2.FONT_HERSHEY_SIMPLEX
+            bottomLeftCornerOfText = (left_txt_corner[0], starty)
+            fontScale = 1
+            fontColor = (255, 255, 255)
+            thickness = 1
+
+            cv2.putText(img, txt, bottomLeftCornerOfText,
+                        font, fontScale, fontColor, thickness)
+
+            starty += 30
+
+    if title_txt_list is not None:
+        font = cv2.FONT_HERSHEY_SIMPLEX
+        bottomLeftCornerOfText = title_txt_corner
+        fontScale = 1
+        fontColor = (255, 255, 255)
+        thickness = 3
+
+        for txt in title_txt_list:
+            cv2.putText(img, txt, bottomLeftCornerOfText,
+                        font, fontScale, fontColor, thickness)
+            bottomLeftCornerOfText = (bottomLeftCornerOfText[0] + 100,
+                                      bottomLeftCornerOfText[1] + 50)
+
+    canvas = cv2.cvtColor(img, cv2.COLOR_RGB2BGR)
+    cv2.imshow(canvas_name, canvas)
```

Comment:
I am wondering if cv2.imshow and cv2.waitKey is appropriate here since it's not quite flexible to determine the wait time.

Reply:
I'm not quite certain about the role of this function. It is kind of specific to putting text at somewhere on the image. If we remove the `imshow`, it becomes a "cv_put_text" function.

Reply:
maybe we can leave the cv2.waitKey to users?

Reply:
just like we leave plt.show() to users

Reply:
Without `waitKey`, would `imshow` still trigger the plotting?

Reply:
Yes, but no window will pop out until you call cv2.waitKey

Reply:
OK that sounds fair.